src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -2357,12 +2357,25 @@ module internal ParseAndCheckFile =
         let tokenizer =
             LexFilter.LexFilter(indentationSyntaxStatus, options.CompilingFSharpCore, Lexer.token lexargs true, lexbuf, false)
 
-        (fun _ -> tokenizer.GetToken())
+        if ct.CanBeCanceled then
+            (fun _ ->
+                ct.ThrowIfCancellationRequested()
+                tokenizer.GetToken())
+        else
+            (fun _ -> tokenizer.GetToken())
 
     let createLexbuf langVersion strictIndentation sourceText =
         UnicodeLexing.SourceTextAsLexbuf(true, LanguageVersion(langVersion), strictIndentation, sourceText)
 
-    let matchBraces (sourceText: ISourceText, fileName, options: FSharpParsingOptions, userOpName: string, suggestNamesForErrors: bool) =
+    let matchBraces
```

Comment:
So what about making `matchBraces` and `parseFile` `Cancellable` instead of manually passing in the token?

Reply:
It's internal and I covered all usages => I guess it does not matter?

Reply:
I just thought it would be more consistent since we use `Cancellable` for this purpose elsewhere.